src/main/Main.hs

```diff
@@ -123,6 +128,12 @@ main = withInterpreterArgs stackProgName $ \args isInterpreter -> do
                                                     commitCount /= ("UNKNOWN" :: String)]
             , [" ", display buildArch]
             ]
+#else
+     let versionString' = concat [ showVersion Meta.version
+                                 , " (git version info disabled)"
```

Comment:
I think you can just omit the ` (git version info disabled)` part.  Similar to when Stack is built using `cabal install stack` (where no git info is available), which also just leaves it out.